sys/kern/src/startup.rs

```diff
@@ -161,25 +127,40 @@ fn safe_start_kernel(
     // As a small optimization, we equip each task with an array of references
     // to RegionDecs, instead of looking them up by index each time. Generate
     // these.
-    let region_tables = alloc.gimme_n(app_header.task_count as usize, |i| {
-        let mut table = [&region_descs[0]; app::REGIONS_PER_TASK];
+
+    // Safety: MaybeUninit<[T]> -> [MaybeUninit<T>] is defined as safe.
+    let region_tables: &mut [[MaybeUninit<&'static app::RegionDesc>; app::REGIONS_PER_TASK];
+             HUBRIS_TASK_COUNT] =
+        unsafe { &mut *(region_tables as *mut _ as *mut _) };
```

Comment:
Out of curiosity, what's the intermediate state in this chain of casts?

Reply:
`&mut MaybeUninit<[T]>`
to `*mut MaybeUninit<[T]>`
to `*mut [MaybeUninit<T>]`
then `&mut [MaybeUninit<T>]`

Reply:
I can document this further if you want, but this is basically a mechanical dance I do to avoid needing transmute in cases like this -- the start and end types pretty much tell the story.